capsules/cyw4343/src/sdpcm.rs

```diff
@@ -0,0 +1,601 @@
+// Licensed under the Apache License, Version 2.0 or the MIT License.
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+// Copyright OxidOS Automotive 2025.
+
+//! CYW4343x SDPCM protocol headers and packet types
+
+use core::ffi::CStr;
+
+use enum_primitive::cast::FromPrimitive;
+use enum_primitive::enum_from_primitive;
+
+use capsules_extra::wifi;
+
+macro_rules! parse {
```

Comment:
It would be great to have a big comment here about what this does and why. It's used heavily in the file, but also pretty hard to read.